### ISSUE TITLE
Add spatial reference support to bounds object

### ISSUE THREAD

**Issue by msmitherdc** [PDAL/PDAL#2295]
This would allow bounds fetching from ept/greyhound via any supported srs rather than requiring native.

**Comment by hobu** [comment]
Let's instead add a `geometry` object that can have an `srs` sub-object instead. If we had a `geometry` object that was a GeoJSON geometry with full support provided by OGR, the driver could reproject and fetch a bounds in native coordinate system as needed.

**Comment by abellgithub** [comment]
From a user perspective, isn't this just about providing a unified syntax?  We already support geometry/SRS with filters.crop, for instance, but the information is provided as separate inputs.

**Comment by hobu** [comment]
> From a user perspective, isn't this just about providing a unified syntax?

Yeah, and also providing a capable object type that stages can utilize. There's Bounds right now that isn't very smart, and Geometry doesn't support assigning `srs` as a sub-object at the moment.
